components/count-badge/count-badge.js

```diff
@@ -134,7 +172,15 @@ class CountBadge extends RtlMixin(LitElement) {
 			numberString = `${'9'.repeat(this.maxDigits)}+`;
 		}
 		return html`
-        	<div class="d2l-count-badge-number">${numberString}</div>`;
+        	<div
+			id="${this._badgeId}"
+			tabindex="${ifDefined(this.tabStop || this.tooltip ? '0' : undefined)}" 
+			role="${ifDefined(this.announceChanges ? 'status' : undefined)}"
+			aria-label="${this.text}">
+				<div class="d2l-count-badge-number" aria-hidden="true">${numberString}</div>
+			${this.tooltip ? html`<d2l-tooltip for="${this._badgeId}">${this.text}</d2l-tooltip>` : html`<span class="d2l-offscreen">"${this.text}"</span>`}
+			</div>
```

Comment:
The tooltip has a mode where it can act as the label for a thing, and when used, it makes using `aria-label` redundant. But of course then when it's not used, you still need the label. One thing's for sure though: you don't need `aria-label` AND the offscreen text.

I'd suggest something like this:

```suggestion
        	<div
			id="${this._badgeId}"
			tabindex="${ifDefined(this.tabStop || this.tooltip ? '0' : undefined)}" 
			role="${ifDefined(this.announceChanges ? 'status' : undefined)}"
			aria-label="${ifDefined(this.tooltip ? undefined : this.text)}">
				<div class="d2l-count-badge-number" aria-hidden="true">${numberString}</div>
			${this.tooltip ? html`<d2l-tooltip for="${this._badgeId}">${this.text}</d2l-tooltip>` : null}
			</div>
```